src/shallow/Snapshot/Serialize/Serializer.lua

```diff
@@ -0,0 +1,247 @@
+local RoactRoot = script.Parent.Parent.Parent.Parent
+
+local ElementKind = require(RoactRoot.ElementKind)
+local Ref = require(RoactRoot.PropMarkers.Ref)
+local Type = require(RoactRoot.Type)
+local Markers = require(script.Parent.Markers)
+local IndentedOutput = require(script.Parent.IndentedOutput)
+
+local Serializer = {}
+
+function Serializer.kind(kind)
+	if kind == ElementKind.Host then
+		return "Host"
+	elseif kind == ElementKind.Function then
+		return "Function"
+	elseif kind == ElementKind.Stateful then
+		return "Stateful"
+	else
+		error(("Cannot serialize ElementKind %q"):format(tostring(kind)))
+	end
+end
```

Comment:
Yes, I removed it from the ShallowWrapper object. Before you could access it with
```lua
wrapper.type.kind
```
The type field would also contain `className`, `functionComponent` or `classComponent` depending on the kind of the component.

Now, I removed it and simply offer a `component` field that is just like the component field from the VirtualNode. Which is a string for host's class name, a function or a table for function or stateful components.

When I create the SnapshotData from the ShallowWrapper, I get back the ElementKind from `ElementKind.fromComponent` in order to serialize it into the ModuleScript. So that way, the source looks like this
```lua
type = {
    kind = ElementKind.Stateful,
    componentName = "CoolComponent",
}
type = {
    kind = ElementKind.Function,
},
type = {
    kind = ElementKind.Host,
    className = "Frame",
},
```

As you can see, it's mostly useful for function components, because I don't have anything meaningful to serialize about their nature than their ElementKind.